client/www/components/dash/explorer/Explorer.tsx

```diff
@@ -690,7 +727,401 @@ export function Explorer({
+                  // Regular single click selection
+                  setCheckedIds((prev) => {
+                    const newCheckedIds = { ...prev };
+                    if (e.target.checked) {
+                      newCheckedIds[row.id] = true;
+                    } else {
+                      delete newCheckedIds[row.id];
+                    }
+                    return newCheckedIds;
+                  });
+                }
+
+                lastSelectedIdRef.current = column.id;
+              }}
+            />
+            {readOnlyNs ? null : (
+              <button
+                className="translate-y-[2px] opacity-0 transition-opacity group-hover:opacity-100"
+                onClick={() => setEditableRowId(row.id)}
+              >
+                <PencilSquareIcon className="h-4 w-4 text-neutral-500 dark:text-neutral-400" />
+              </button>
+            )}
+          </div>
+        );
+      },
+    });
+
+    selectedNamespace?.attrs?.forEach((attr) => {
+      result.push({
+        id: attr.id + attr.name,
+        header: attr.name,
+        enableSorting: true,
+        enableResizing: true,
+        accessorFn: (row) => row[attr.name],
+        meta: {
+          sortable: attr.sortable || attr.name === 'id',
+          copyable: true,
+          isLink: attr.type === 'ref',
+          attr,
+          disablePadding: attr.namespace === '$files' && attr.name === 'url',
+        } satisfies TableColMeta,
+        cell: (info) => {
+          if (
+            info.row.original[attr.name] === null ||
+            info.row.original[attr.name] === undefined
+          ) {
+            return <div className="h-1">-</div>;
+          }
+          if (attr.type === 'ref') {
+            const linkCount = info.row.original[attr.name].length;
+            return (
+              <div
+                className={cn(
+                  'h-1 translate-y-[2px]',
+                  linkCount < 1 && 'opacity-50',
+                )}
+              >
+                {linkCount} link{linkCount === 1 ? '' : 's'}
+              </div>
+            );
+          }
+
+          if (attr.namespace === '$files') {
+            if (attr.name === 'url') {
+              return (
+                <a
+                  className="h-full w-full pl-2 align-middle text-xs font-bold underline hover:text-black dark:hover:text-white"
+                  href={info.row.original['url'] as string}
+                  target="_blank"
+                >
+                  View File
+                </a>
+              );
+            } else if (attr.name === 'size') {
+              return formatBytes(info.row.original[attr.name]);
+            }
+          }
+
+          if (attr.checkedDataType === 'boolean') {
+            return info.row.original[attr.name] ? 'true' : 'false';
+          }
+          if (attr.checkedDataType === 'date') {
+            return new Date(info.row.original[attr.name]).toLocaleString();
```

Comment:
Should this use `coerceToDates` from dates.ts or do we know for sure that it's date-able already?